include/sir/helpers.h

```diff
@@ -333,6 +333,15 @@ uint32_t FNV32_1a(const uint8_t* data, size_t len) {
     return hash;
 }
 
+/**
+ * Wrapper for snprintf when truncation is intended.
+ */
+# define snprintf_truncate(dst, size, ...) \
+    do { \
+      volatile size_t n = size; \
+      (void)snprintf (dst, n, __VA_ARGS__); \
```

Comment:
No space before parens please. And why do you have to make a volatile copy of size?

Reply:
The GCC authors provided this macro as the preferred way to silence this warning when truncation is expected or desired.  There is a link at https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89312 that goes more into it.  There isn't yet any alternative I've seen that works in all cases.